game-of-life-native/Game of Life/GameOfLifeViewController.swift

```diff
@@ -11,59 +11,104 @@ import Foundation
+
     var onControlButtonClicked: () -> KGOLStdlibUnit = {
         return KGOLStdlibUnit()
     }
-    var onPatternSelected: (KGOLPatternEntity) -> KGOLStdlibUnit = {_ in
-        return  KGOLStdlibUnit()
+    var onPatternSelected: (KGOLPatternEntity) -> KGOLStdlibUnit = { _ in
+        return KGOLStdlibUnit()
     }
-    
+
     required init?(coder aDecoder: NSCoder) {
+        let cellMatrix = KGOLListBasedMatrix(width: 20, height: 20, seeds: NSArray() as! [Any])
+        let boardEntity = KGOLSimulationBoardEntity(cellMatrix: cellMatrix)
+        let loop = SwiftGameLoop() as KGOLGameLoop
+        let model = KGOLBoardModelImpl(initialBoard: boardEntity, gameLoop: loop)
+        boardPresenter = KGOLBoardPresenter(boardModel: model)
+
         super.init(coder: aDecoder)
+        controlButton.addTarget(self, action: #selector(controlButtonAction), for: .touchUpInside)
+        patternSelectionButton.addTarget(self, action: #selector(patternSelectionButtonAction), for: .touchUpInside)
     }
-    
+
+    private func setupLayout() {
+        let yOffset: CGFloat = CGFloat(35)
+
+        controlButton.frame = CGRect(
+                x: 0,
+                y: yOffset,
+                width: self.view.bounds.width / 2,
+                height: yOffset)
+
+        patternSelectionButton.frame = CGRect(
+                x: self.view.bounds.width / 2,
+                y: yOffset,
+                width: self.view.bounds.width / 2,
+                height: yOffset)
+        patternSelectionButton.setTitle("Choose a Pattern", for: .normal)
+
+        boardView.backgroundColor = .black
+        boardView.frame = CGRect(
+                x: 0,
+                y: controlButton.bounds.height + yOffset,
+                width: self.view.bounds.width,
+                height: self.view.bounds.width)
+    }
+
     func renderControlButtonLabel(controlButtonLabel: String) {
-        controllButton.setTitle(controlButtonLabel, for: .normal)
+        controlButton.setTitle(controlButtonLabel, for: .normal)
     }
-    
+
     func renderPatternSelectionVisibility(visibility: Bool) {
-        
+        patternSelectionButton.isHidden = !visibility
     }
-    
+
     func renderBoardWith(boardViewInput: KGOLBoardViewInput) {
         boardView.renderBoard(boardViewInput: boardViewInput)
     }
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        if (appPresenter == nil){
-            let model = KGOLAppModel()
-            appPresenter = KGOLAppPresenter(model: model)
-            controllButton.backgroundColor = .green
-            controllButton.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
-            self.view.addSubview(controllButton)
-            
-            boardView.backgroundColor = .blue
-            self.view.addSubview(boardView)
-        }
-        
-        appPresenter?.bind(view: self)
-        boardView.willAppear()
+        view.addSubview(controlButton)
```

Comment:
might be worth it to create a function `setupViews()` that you call here and add the button `addTarget` and the functions for `backgroundColor` and `setTitle`